File: 2581-divide-players-into-teams-of-equal-skill/2581-divide-players-into-teams-of-equal-skill.cpp
class Solution {
public:
    long long dividePlayers(std::vector<int>& skill) {
    std::sort(skill.begin(), skill.end());
    
    int left = 0;
    int right = skill.size() - 1;
    long sum = 0;
    int targetSum = skill[left] + skill[right];

    while (left < right) {
        if (skill[left] + skill[right] != targetSum) {
            std::cout << skill[left] + skill[right] << " " << targetSum << std::endl;
            return -1;
        }
        sum += skill[left] * skill[right];
        left++;
        right--;
    }

    return sum;
}


};

File: 2581-divide-players-into-teams-of-equal-skill/2581-divide-players-into-teams-of-equal-skill.js
/**
 * @param {number[]} skill
 * @return {number}
 */

var dividePlayers = function (skill) {
    skill.sort((a,b) => a-b)
    
    let left = 0;
    let right = skill.length - 1;
    let sum = 0;
    let targetSum = skill[left] + skill[right];

    while (left < right) {
        if (skill[left] + skill[right] !== targetSum) {
            console.log(skill[left] + skill[right]  , targetSum)
            return -1;
        }
        sum += skill[left] * skill[right];
        left++;
        right--;
    }

    return sum;
};